hushline/routes.py

```diff
@@ -258,41 +267,53 @@ def login() -> Response | str:
                     session["is_admin"] = user.is_admin
 
                     if user.totp_secret:
-                        return redirect(url_for("verify_2fa_login"))
+                        return redirect(url_for("verify_2fa_login")), 302
                     else:
                         session["2fa_verified"] = True
-                        return redirect(url_for("inbox", username=user.primary_username))
+                        return redirect(url_for("inbox", username=user.primary_username)), 302
                 else:
                     flash("⛔️ Invalid username or password")
+                    return render_template("login.html", form=form), 401
             else:
                 flash("⛔️ Invalid form data")
-        return render_template("login.html", form=form)
+                return render_template("login.html", form=form), 400
+        return render_template("login.html", form=form), 200
 
     @app.route("/verify-2fa-login", methods=["GET", "POST"])
     @limiter.limit("120 per minute")
     def verify_2fa_login() -> Response | str:
-        # Redirect to login if user is not authenticated
+        # Initialize form at the beginning to ensure it is available throughout the function
+        form = TwoFactorForm()
```

Comment:
you can init the form where it was before since it's not used in the first `if` block

Reply:
👍